Fixes for new module (#6)
sanddev-com/cookieconsent: README.md

[![MIT License][li]][ll]

## Additional instructions for this fork from [Sandstream Development](https://github.com/sandstreamdev)

This fork introduces additional categories dialog which is shown after clicking the _Customize settings_ button on the initial cookie consent dialog. It also adds few additional features to handle some GDPR concerns - please see below.

Check [GDPR example](./examples/example-gdpr.html) to see it in action.

<p align="center"><img width="800" src="./images/cc-dialog.png" /></p>

<p align="center"><img width="800" src="./images/categories-dialog.png" /></p>

### Installation

```bash
npm install @sandstreamdev/cookieconsent
# or via yarn
yarn add @sandstreamdev/cookieconsent
```

### Usage

#### Module
```javascript
import { CookieConsent } from '@sandstreamdev/cookieconsent';
// or
const { CookieConsent } = require('@sandstreamdev/cookieconsent');
```

#### Classic
```javascript

> const CC = window.CookieConsent

Reviewer: ```suggestion
const CC = window.CookieConsent;
```

[![MIT License][li]][ll]

## Additional instructions for this fork from [Sandstream Development](https://github.com/sandstreamdev)

> This fork introduces additional categories dialog which is show after clicking 'Customize settings' button on initial cookies consent dialog. There are some more additional features to handle GDPR - see below.

Reviewer: ```suggestion
This fork introduces additional categories dialog which is shown after clicking the Customize settings button on the initial cookie consent dialog. There are some more additional features to handle more GDPR concerns - please see below.
```

Check [GDPR example](./examples/example-gdpr.html) to see it in action.

> <p align="center"><img width="800" src="images/cc-dialog.png"></a></p>

Reviewer: Misclosed anchors?

```suggestion
<p align="center"><img width="800" src="images/cc-dialog.png"></p>
```

Author: nice catch :) copy&paste issue ;)

> <p align="center"><img width="800" src="images/categories-dialog.png"></a></p>

Reviewer: ```suggestion
<p align="center"><img width="800" src="images/categories-dialog.png"></p>
```

### Additional configuration options

#### Categories

> For showing only specific categories following options needs to be set:

Reviewer: ```suggestion
For showing only specific categories, the following options need to be set:
```

> ```

Reviewer: ```suggestion
```javascript
```

const cc = new CC({
  //...options,
  showCategories: {
    [CC.UNCATEGORIZED]: false,
    [CC.ESSENTIAL]: true,
    [CC.PERSONALIZATION]: false,
    [CC.ANALYTICS]: true,
    [CC.MARKETING]: false
  }
})
```

> Above example will show only essential and analytics categories.

Reviewer: ```suggestion
The above example will only show essential and analytics categories.
```

#### Policies links

> To set policies links mentioned in dialogs use following options:

Reviewer: ```suggestion
To wire the policies links mentioned in dialogs, please use the following options:
```

> ```

Reviewer: Please toggle on syntax highlighting.

```suggestion
```javascript
```

const cc = new CC({
  //...options,
  content: {
    privacyPolicyLink: '/privacy-policy',
    cookiePolicyLink: '/cookie-policy'
  }
})
```

#### Consent settings

> As in GDPR you need to provide functionality to let the user change consents `consentSettingsElementId` was added. It is id of element on the page that will reopen categories dialog after clicking it.

Reviewer: ```suggestion
As in GDPR, you need to provide the functionality to let the user change consents, so the `consentSettingsElementId` was added. It is the id of the element on the page that will reopen the categories dialog after clicking it.
```

```
const cc = new CC({
  //...options,
  consentSettingsElementId: 'btn-revokeChoice'
});
```

> #### Replace default `Read more` texts for categories

Reviewer: ```suggestion
#### How to replace the default `Read more` texts for categories?
```

> All or some subset of categories' read more text can be changed.

Reviewer: ```suggestion
All or the desired subset of categories' read more text can be changed.
```

```
const cc = new CC({
  //...options,
  content: {
    categoryAnalytics: 'My new awesome text for analytics category',
    categoryEssential: 'My new awesome text for essential category',
    categoryMarketing: 'My new awesome text for marketing category',
    categoryPersonalization: 'My new awesome text for personalization category',
    categoryUncategorized: 'My new awesome text for uncategorized category'
  }
});
```

> #### How to set the Google Tag Manager (GTM) cookie only if user accepted `Analytics` cookies?

Reviewer: Please avoid using backticks and code blocks for non-code related words. Use emphasis (double **) or double underlines (or none if not really needed).

```suggestion
#### How to set the Google Tag Manager (GTM) cookie only if the user has accepted analytics cookies?
```

Check [GDPR example](./examples/example-gdpr.html) to see the solution.

> It handles also scenario when user revoke consent.

Reviewer: ```suggestion
It also handles scenarios where users revoke their consent.
```

> #### Can be styled fully (or almost ;)) with css

Reviewer: ```suggestion
#### Styling with CSS
```

Styled version used in one of our projects:

> <p align="center"><img width="800" src="images/cc-dialog-styled.png"></a></p>

Reviewer: ```suggestion
<p align="center"><img width="800" src="./images/cc-dialog-styled.png"></a></p>
```

> <p align="center"><img width="800" src="images/categories-dialog-styled.png"></a></p>

Reviewer: ```suggestion
<p align="center"><img width="800" src="./images/categories-dialog-styled.png"></a></p>
```

> ## And that's all from [Sandstream Development](https://github.com/sandstreamdev). Enjoy!

Reviewer: I think this should not be heading :)
